module/VuFind/src/VuFind/View/Helper/Bootstrap3/Component.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Component view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Bootstrap3;
```

Comment:
Is there a strong reason to put this in Bootstrap3 rather than Root? If we plan to use different template storage patterns on a per-theme basis, that would make sense, but if we want to set a global convention, it's probably cleaner to stick to Root so we don't have to move it around later.

Reply:
The reason here is that this dropdown still relies on the Bootstrap library JS. A framework agnostic would definitely go in root.

Reply:
A template dependency on the Bootstrap library would explain why the relevant _ui/component/x.phtml file should go in the bootstrap3 theme... but I think this helper is entirely theme/framework-agnostic and could move to Root. I think it's okay to separate the helper (which defines how component templates are located and loaded) from the templates themselves (which could easily have different implementations in different themes... or even entirely different component sets...). As long as, within any given theme, we only render components defined within that theme, I think we're in good shape, and the component helper doesn't need to care. Am I making any sense? :-)

Reply:
I misunderstood and now agree.